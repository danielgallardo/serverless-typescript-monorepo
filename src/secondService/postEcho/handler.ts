import middy from 'middy';
import {NormalizedEvent} from '../../@types';
import {apiRequestRoutine} from '../../common/middlewares/apiRequestRoutine';
import {eventValidator} from "../../common/middlewares/eventValidator";
import {logRoutine} from '../../common/middlewares/logRoutine';
import {Joi} from '../../common/validation';
import {postEcho} from './postEcho';

interface Event extends NormalizedEvent {
  body: {
    message: string
  };
}

const schema = {
  body: Joi.object().keys({
    message: Joi.string().required()
  })
};

const handler = (event: Event) => {
  return postEcho({
    message: event.body.message
  });
};

export default middy(handler)
  .use(apiRequestRoutine())
  .use(eventValidator(schema))
  .use(logRoutine());
