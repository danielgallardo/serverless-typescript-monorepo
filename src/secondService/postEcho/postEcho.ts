import {getEnv} from '../../common/validation';

type Params = {
  message: string;
}

export const postEcho = async ({message}: Params) => {
  const currentStage = getEnv('CURRENT_STAGE');
  const currentRegion = getEnv('AWS_REGION');
  return {
    currentStage,
    currentRegion,
    message
  }
};
