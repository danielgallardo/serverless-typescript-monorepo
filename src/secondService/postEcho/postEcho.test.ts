import {postEcho} from './postEcho';

process.env.CURRENT_STAGE = 'dev';
process.env.AWS_REGION = 'eu-west-1';

describe('postEcho', () => {
  it('should return correct response', () => {
    return expect(postEcho({message: 'Some message'})).resolves.toMatchSnapshot();
  });
});
