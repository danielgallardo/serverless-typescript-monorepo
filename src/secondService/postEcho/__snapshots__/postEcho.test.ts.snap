// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`postEcho should return correct response 1`] = `
Object {
  "currentRegion": "eu-west-1",
  "currentStage": "dev",
  "message": "Some message",
}
`;
